pkg/providers/subnet/subnet.go

```diff
@@ -182,26 +212,26 @@ func (p *DefaultProvider) UpdateInflightIPs(createFleetInput *ec2.CreateFleetInp
 	// Find the subnets that were included in the input but not chosen by Fleet, so we need to add the inflight IPs back to them
 	subnetIDsToAddBackIPs, _ := lo.Difference(fleetInputSubnets, fleetOutputSubnets)
 
-	// Aggregate all the cached subnets
-	cachedSubnets := lo.UniqBy(lo.Flatten(lo.MapToSlice(p.cache.Items(), func(_ string, item cache.Item) []*ec2.Subnet {
-		return item.Object.([]*ec2.Subnet)
-	})), func(subnet *ec2.Subnet) string { return *subnet.SubnetId })
+	// Aggregate all the cached subnets ip address count
+	cachedIPAddressCount := lo.MapEntries(p.availableIPAddressCache.Items(), func(k string, v cache.Item) (string, int64) {
+		return k, v.Object.(int64)
+	})
 
 	// Update the inflight IP tracking of subnets stored in the cache that have not be synchronized since the initial
 	// deduction of IP addresses before the instance launch
-	for _, cachedSubnet := range cachedSubnets {
-		if !lo.Contains(subnetIDsToAddBackIPs, *cachedSubnet.SubnetId) {
+	for cachedSubnetID, cachedIPAddressCount := range cachedIPAddressCount {
```

Comment:
Should we rename this map? You're calling the key the same thing as the map, which is going to get confusing